### ISSUE TITLE
Expand DLL search in setup.py

### ISSUE THREAD

**Issue by jtniehof** [spacepy/spacepy#688]
On the Windows build, the mingw32 runtime libraries get copied into the spacepy build in case they go away (and also to support the binary wheel). Right now it only searches PATH. That's not really the preferred Windows DLL search method anymore and it may be causing problems. The installer should be updated to search via other means as well, probably using [find_library](https://docs.python.org/3/library/ctypes.html#finding-shared-libraries) (although there might be a better way to get the DLL search path).

As a workaround, adding the directory with the ming DLLS to PATH _should_ work.

### Minimal example to reproduce issue:
It's not clear under what circumstances this is sufficient, but @rebeccaringuette noted (and thanks!)

```
pip install ./spacepy
```

### Error message/Traceback:

```
Processing c:\users\rringuet\pyhcs3\spacepy

  Installing build dependencies ... done

  Getting requirements to build wheel ... done

  Preparing metadata (pyproject.toml) ... done

Requirement already satisfied: matplotlib>=3.1 in c:\users\rringuet\anaconda3\envs\pyhcs3\lib\site-packages (from spacepy==0.5.0a0) (3.7.1)

Requirement already satisfied: scipy>=1.0 in c:\users\rringuet\anaconda3\envs\pyhcs3\lib\site-packages (from spacepy==0.5.0a0) (1.10.1)

Requirement already satisfied: python-dateutil>=2.1 in c:\users\rringuet\anaconda3\envs\pyhcs3\lib\site-packages (from spacepy==0.5.0a0) (2.8.2)

Requirement already satisfied: h5py>=2.10 in c:\users\rringuet\anaconda3\envs\pyhcs3\lib\site-packages (from spacepy==0.5.0a0) (3.8.0)

Requirement already satisfied: numpy>=1.15.1 in c:\users\rringuet\anaconda3\envs\pyhcs3\lib\site-packages (from spacepy==0.5.0a0) (1.24.3)

Requirement already satisfied: cycler>=0.10 in c:\users\rringuet\anaconda3\envs\pyhcs3\lib\site-packages (from matplotlib>=3.1->spacepy==0.5.0a0) (0.11.0)

Requirement already satisfied: contourpy>=1.0.1 in c:\users\rringuet\anaconda3\envs\pyhcs3\lib\site-packages (from matplotlib>=3.1->spacepy==0.5.0a0) (1.0.7)

Requirement already satisfied: kiwisolver>=1.0.1 in c:\users\rringuet\anaconda3\envs\pyhcs3\lib\site-packages (from matplotlib>=3.1->spacepy==0.5.0a0) (1.4.4)

Requirement already satisfied: pillow>=6.2.0 in c:\users\rringuet\anaconda3\envs\pyhcs3\lib\site-packages (from matplotlib>=3.1->spacepy==0.5.0a0) (9.5.0)

Requirement already satisfied: packaging>=20.0 in c:\users\rringuet\anaconda3\envs\pyhcs3\lib\site-packages (from matplotlib>=3.1->spacepy==0.5.0a0) (23.1)

Requirement already satisfied: pyparsing>=2.3.1 in c:\users\rringuet\anaconda3\envs\pyhcs3\lib\site-packages (from matplotlib>=3.1->spacepy==0.5.0a0) (3.0.9)

Requirement already satisfied: fonttools>=4.22.0 in c:\users\rringuet\anaconda3\envs\pyhcs3\lib\site-packages (from matplotlib>=3.1->spacepy==0.5.0a0) (4.39.3)

Requirement already satisfied: six>=1.5 in c:\users\rringuet\anaconda3\envs\pyhcs3\lib\site-packages (from python-dateutil>=2.1->spacepy==0.5.0a0) (1.16.0)

Building wheels for collected packages: spacepy

  Building wheel for spacepy (pyproject.toml) ... error

  error: subprocess-exited-with-error

 

  × Building wheel for spacepy (pyproject.toml) did not run successfully.

  │ exit code: 1

  ╰─> [58 lines of output]

      running bdist_wheel

      running build

      C compiler: gcc -mdll -O -Wall

 

      creating build

      creating build\temp.win-amd64-cpython-310

      creating build\temp.win-amd64-cpython-310\spacepy

      creating build\temp.win-amd64-cpython-310\spacepy\libspacepy

      compile options: '-c'

      gcc -mdll -O -Wall -c spacepy\libspacepy\randomkit.c -o build\temp.win-amd64-cpython-310\spacepy\libspacepy\randomkit.o

      gcc -mdll -O -Wall -c spacepy\libspacepy\cn_solver.c -o build\temp.win-amd64-cpython-310\spacepy\libspacepy\cn_solver.o

      gcc -mdll -O -Wall -c spacepy\libspacepy\poppy.c -o build\temp.win-amd64-cpython-310\spacepy\libspacepy\poppy.o

      gcc -mdll -O -Wall -c spacepy\libspacepy\ctrace2d.c -o build\temp.win-amd64-cpython-310\spacepy\libspacepy\ctrace2d.o

      gcc -mdll -O -Wall -c spacepy\libspacepy\toolbox.c -o build\temp.win-amd64-cpython-310\spacepy\libspacepy\toolbox.o

     creating build\lib

      creating build\lib\spacepy

      gcc -shared -s build\temp.win-amd64-cpython-310\spacepy\libspacepy\cn_solver.o build\temp.win-amd64-cpython-310\spacepy\libspacepy\ctrace2d.o build\temp.win-amd64-cpython-310\spacepy\libspacepy\poppy.o build\temp.win-amd64-cpython-310\spacepy\libspacepy\randomkit.o build\temp.win-amd64-cpython-310\spacepy\libspacepy\toolbox.o -lm -o build\lib\spacepy\libspacepy.dll.a

      Traceback (most recent call last):

        File "C:\Users\rringuet\Anaconda3\envs\PyHCs3\lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 353, in <module>

          main()

        File "C:\Users\rringuet\Anaconda3\envs\PyHCs3\lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 335, in main

          json_out['return_val'] = hook(**hook_input['kwargs'])

        File "C:\Users\rringuet\Anaconda3\envs\PyHCs3\lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 251, in build_wheel

          return _build_backend().build_wheel(wheel_directory, config_settings,

        File "C:\Users\rringuet\AppData\Local\Temp\6\pip-build-env-cm4azeys\overlay\Lib\site-packages\setuptools\build_meta.py", line 416, in build_wheel

          return self._build_with_temp_dir(['bdist_wheel'], '.whl',

        File "C:\Users\rringuet\AppData\Local\Temp\6\pip-build-env-cm4azeys\overlay\Lib\site-packages\setuptools\build_meta.py", line 401, in _build_with_temp_dir

          self.run_setup()

        File "C:\Users\rringuet\AppData\Local\Temp\6\pip-build-env-cm4azeys\overlay\Lib\site-packages\setuptools\build_meta.py", line 487, in run_setup

          super(_BuildMetaLegacyBackend,

        File "C:\Users\rringuet\AppData\Local\Temp\6\pip-build-env-cm4azeys\overlay\Lib\site-packages\setuptools\build_meta.py", line 338, in run_setup

          exec(code, locals())

        File "<string>", line 971, in <module>

        File "C:\Users\rringuet\AppData\Local\Temp\6\pip-build-env-cm4azeys\overlay\Lib\site-packages\numpy\distutils\core.py", line 169, in setup

          return old_setup(**new_attr)

        File "C:\Users\rringuet\AppData\Local\Temp\6\pip-build-env-cm4azeys\overlay\Lib\site-packages\setuptools\__init__.py", line 107, in setup

          return distutils.core.setup(**attrs)

        File "C:\Users\rringuet\AppData\Local\Temp\6\pip-build-env-cm4azeys\overlay\Lib\site-packages\setuptools\_distutils\core.py", line 185, in setup

          return run_commands(dist)

        File "C:\Users\rringuet\AppData\Local\Temp\6\pip-build-env-cm4azeys\overlay\Lib\site-packages\setuptools\_distutils\core.py", line 201, in run_commands

          dist.run_commands()

        File "C:\Users\rringuet\AppData\Local\Temp\6\pip-build-env-cm4azeys\overlay\Lib\site-packages\setuptools\_distutils\dist.py", line 969, in run_commands

          self.run_command(cmd)

        File "C:\Users\rringuet\AppData\Local\Temp\6\pip-build-env-cm4azeys\overlay\Lib\site-packages\setuptools\dist.py", line 1244, in run_command

          super().run_command(command)

        File "C:\Users\rringuet\AppData\Local\Temp\6\pip-build-env-cm4azeys\overlay\Lib\site-packages\setuptools\_distutils\dist.py", line 988, in run_command

          cmd_obj.run()

        File "C:\Users\rringuet\AppData\Local\Temp\6\pip-build-env-cm4azeys\overlay\Lib\site-packages\wheel\bdist_wheel.py", line 343, in run

          self.run_command("build")

        File "C:\Users\rringuet\AppData\Local\Temp\6\pip-build-env-cm4azeys\overlay\Lib\site-packages\setuptools\_distutils\cmd.py", line 318, in run_command

          self.distribution.run_command(command)

        File "C:\Users\rringuet\AppData\Local\Temp\6\pip-build-env-cm4azeys\overlay\Lib\site-packages\setuptools\dist.py", line 1244, in run_command

          super().run_command(command)

        File "C:\Users\rringuet\AppData\Local\Temp\6\pip-build-env-cm4azeys\overlay\Lib\site-packages\setuptools\_distutils\dist.py", line 988, in run_command

          cmd_obj.run()

        File "<string>", line 752, in run

        File "<string>", line 816, in copy_dlls

      RuntimeError: Can't locate runtime libraries.

      [end of output]

 

  note: This error originates from a subprocess, and is likely not a problem with pip.

  ERROR: Failed building wheel for spacepy

Failed to build spacepy

ERROR: Could not build wheels for spacepy, which is required to install pyproject.toml-based projects
```

The `<string>` there is setup.py

### OS, Python version, and dependency version information:
Python 3.10, Windows 10 22H2
MPL 3.7.1, scipy 1.10.1, numpy 1.24.3 (not that it really matters in this case)

### Version of SpacePy
Current git checkout (main)

**Comment by rebeccaringuette** [comment]
Python version: 3.10.30
Windows 10 version: 22H2 (OS Build 19045.2846)

**Comment by jtniehof** [comment]
> Python version: 3.10.30 Windows 10 version: 22H2 (OS Build 19045.2846)

Thanks. 3.10 is ending at 3.10.11, so not sure about that subminor version, but it's probably not relevant.

**Comment by rebeccaringuette** [comment]
Oops. Typo corrected. Python 3.10.10.

**Comment by jtniehof** [comment]
I did some digging and find_library actually searches PATH as well. So there's something else weird going on and we'll look into it this afternoon.

I guess another approach would be to search PATH for gfortan and use that to try and identify the mingw installation and go searching for the dlls there.

**Comment by jtniehof** [comment]
Root cause of the problem was that the installer assumed everything in the path was at least 4 characters long (oops). It also looks like the set of DLLs is quite different between versions of mingw32 so loosened those back up a bit. These fixes are in #681.
